### PR TITLE
Feature/improve generated syntax message

### PR DESCRIPTION
**Pull request by DorAbu**
Improved "Block" component "validation" and "use_black" methods in cases that the syntax structure is invalid

✨ Created syntax validator

### CODE REVIEW

File: codemate/block.py [hunk 1]
```diff
@@ -330,26 +331,34 @@ def use_black(self) -> str:
...
-            InputError: When the generated Python code isn't valid by black.
+            InputError: When the generated Python 3 code isn't valid.
         """
         try:
-            self.use_black()
-        except black.InvalidInput as error:
-            raise InputError(error) from None
-        else:
-            return True
+            validator.validate(self.syntax(), raise_error=True)
```
Comment (DavidMeu) [inline codemate/block.py:356]: Don't you want to pass `raise_error` from the input?

File: codemate/validator.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    lines = syntax.split("\n")
+    start = max(error.lineno - ERROR_CONTEXT_PADDING - 1, 0)
+    end = min(error.lineno + ERROR_CONTEXT_PADDING, len(lines))
+    for i, line_number in enumerate(range(start, end), start=start):
+        lines[i] = f"{str(line_number + 1).ljust(len(str(end)) + 1, ' ')}|{lines[i]}"
+    exc_index = error.lineno - 1
+    lines[exc_index] = lines[exc_index] + ERROR_POST_FIX
+    err_lines = "\n".join(lines[start:end])
+    exc_name = getattr(type(error), "__name__", "SyntaxError")
+    return InputError(
```
Comment (DavidMeu) [inline codemate/validator.py:21]: Think as for `errors` they should always be raised not in return statements.

File: codemate/validator.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import ast
+
+from codemate.exceptions import InputError
+
+ERROR_CONTEXT_PADDING = 2
+ERROR_POST_FIX = "   <-- 🔍 seems the error is around here"
+
+
+def _format_error(syntax: str, error: SyntaxError) -> InputError:
```
Comment (DavidMeu) [inline codemate/validator.py:9]: Maybe handle more exceptions (```IndexError```..) here with `try/except` block.

**Review by DavidMeu — APPROVED** [review]
I approved,
But see my attached comments 😃

File: codemate/block.py (hunk 1 above)
Comment (DorAbu) [reply codemate/block.py:356]: Correct my mistake

Comment (DorAbu) [reply codemate/block.py:356]: We want always to raise the error.
If the syntax is invalid and the user doesn't want that we will raise the error, we catch it and returns False.

File: codemate/validator.py (hunk 3 above)
Comment (DorAbu) [reply codemate/validator.py:9]: IndexError can't be raised in the syntax generation, in future we can add a tool that will search logical errors in the code

File: codemate/validator.py (hunk 2 above)
Comment (DorAbu) [reply codemate/validator.py:21]: In this function, we are creating the error object, in the upper function we raise it

**Pull request merged**